native/src/components/TtsPlayer.tsx

```diff
@@ -0,0 +1,344 @@
+    () => (content ? [title, ...extractSentencesFromHtml(content, languageCode)] : []),
+    [content, title, languageCode],
+  )
+  const isPersian = languageCode === 'fa'
+  const maxTitle = 20
+  const isTitleLong = title.length > maxTitle ? title.substring(0, maxTitle).concat('...') : title || t('readAloud')
+
+  const initializeTts = useCallback((): void => {
+    Tts.getInitStatus().catch(async error => {
+      reportError(`Tts-Error: ${error.code}`)
+      if (error.code === 'no_engine') {
+        await Tts.requestInstallEngine().catch((e: string) => reportError(`Failed to install tts engine: : ${e}`))
+      }
+    })
+  }, [])
+
+  useEffect(() => {
+    initializeTts()
+
+    Tts.addEventListener('tts-progress', () => setIsPlaying(true))
+    Tts.addEventListener('tts-cancel', () => setIsPlaying(false))
+    Tts.addEventListener('tts-finish', () => {
+      if (sentenceIndex < sentences.length - 1) {
+        setSentenceIndex(prev => prev + 1)
+      } else {
+        setIsPlaying(false)
+        setSentenceIndex(0)
+      }
+    })
+
+    return () => {
+      Tts.removeAllListeners('tts-finish')
+      Tts.removeAllListeners('tts-progress')
+      Tts.removeAllListeners('tts-cancel')
+    }
+  }, [initializeTts, sentenceIndex, sentences.length])
+
+  useEffect(() => {
+    if (isPlaying && sentences.length > 0) {
+      Tts.setDefaultLanguage(languageCode)
+      const percentage = 100
+      Tts.speak(String(sentences[sentenceIndex]), {
+        androidParams: {
+          KEY_PARAM_PAN: 0,
+          KEY_PARAM_VOLUME: volume / percentage,
+          KEY_PARAM_STREAM: 'STREAM_MUSIC',
+        },
+        iosVoiceId: '',
+        rate: 1,
+      })
+    }
+    if (sentences.length === 0) {
+      // if there is no sentences (null) just reset for future use
+      setSentenceIndex(0)
+      setExpandPlayer(false)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isPlaying, languageCode, sentenceIndex, sentences])
+
+  useEffect(() => {
+    // Stop tts if not in foreground
+    const subscription = AppState.addEventListener('change', nextAppState => {
+      if (nextAppState === 'inactive' || nextAppState === 'background') {
+        Tts.stop()
+      }
+    })
+
+    return () => {
+      subscription.remove()
+    }
+  }, [])
+
+  const stopTts = async () => {
+    await Tts.stop()
+    const TTS_STOP_DELAY = 100 // delay to make sure tts is stopped
+    await new Promise(resolve => {
+      setTimeout(resolve, TTS_STOP_DELAY)
+    })
+  }
+  const startReading = () => {
+    if (!isPersian && sentences.length > 0) {
+      // Persian not supported
+      setIsPlaying(true)
+    } else {
+      setIsPlaying(false)
+    }
+  }
+
+  const pauseReading = () => {
+    Tts.stop()
+    setIsPlaying(false)
+  }
+
+  const handleBackward = async () => {
+    await stopTts()
+    setSentenceIndex(prev => Math.max(0, prev - 1)) // it return the bigger number so no negative values
```

Comment:
:wrench: Please don't abbreviate words like this. Also, I think `index` is more clear.
```suggestion
    setSentenceIndex(index => Math.max(0, index - 1))
```